utils/index.ts

```diff
@@ -215,3 +218,29 @@ export function getImageResizeURL (url: string, { width = 300 }: { width?: numbe
   const { LIKE_CO_STATIC_ENDPOINT } = useRuntimeConfig().public
   return `${LIKE_CO_STATIC_ENDPOINT}/thumbnail/?url=${encodeURIComponent(url)}&width=${width}`
 }
+
+export function fileToArrayBuffer (file: Blob): Promise<string | ArrayBuffer | null> {
+  return new Promise((resolve) => {
+    const reader = new FileReader()
+    reader.onload = () => resolve(reader.result)
+    reader.readAsArrayBuffer(file)
+  })
+}
```

Comment:
If we always cast the result as `ArrayBuffer` when we use this, maybe we should make sure result is `ArrayBuffer` in this function instead, esp checking for `null` and avoiding `string` result

Reply:
[🥅 Improve error handling in fileToArrayBuffer](https://github.com/likecoin/nft-book-press/pull/290/commits/ca2d356b74ce6f3e27d572203b30838adef365d9)